EFCoreRelationshipsPracticeTest/ServiceTest/CompanyServiceTest.cs

```diff
@@ -0,0 +1,179 @@
+        {
+        }
+
+        private CompanyDbContext GetCompanyDbContext()
+        {
+            var scope = Factory.Services.CreateScope();
+            var scopedService = scope.ServiceProvider;
+            CompanyDbContext context = scopedService.GetRequiredService<CompanyDbContext>();
+            return context;
+        }
+
+        [Fact]
+        public async Task Should_create_company_through_service()
+        {
+            //given
+            var context = GetCompanyDbContext();
+            CompanyDto companyDto = new CompanyDto();
+            companyDto.Name = "IBM";
+            companyDto.EmployeesDto = new List<EmployeeDto>()
+            {
+                new EmployeeDto()
+                {
+                    Name = "Tom",
+                    Age = 19,
+                }
+            };
+            companyDto.ProfileDto = new ProfileDto()
+            {
+                RegisteredCapital = 10000,
+                CertId = "180"
+            };
+            CompanyService companyService = new CompanyService(context);
+
+            //when
+            await companyService.AddCompany(companyDto);
+
+            //then
+            Assert.Equal(1, context.Companies.Count());
+        }
+
+        [Fact]
+        public async Task Should_get_company_by_Id_through_service()
+        {
+            //given
+            var context = GetCompanyDbContext();
+            CompanyDto companyDto = new CompanyDto();
+            companyDto.Name = "IBM";
+            companyDto.EmployeesDto = new List<EmployeeDto>()
+            {
+                new EmployeeDto()
+                {
+                    Name = "Tom",
+                    Age = 19,
+                }
+            };
+            companyDto.ProfileDto = new ProfileDto()
+            {
+                RegisteredCapital = 10000,
+                CertId = "180"
+            };
+            CompanyService companyService = new CompanyService(context);
+            var id = await companyService.AddCompany(companyDto);
+            Assert.Equal(1, context.Companies.Count());
+
+            //when
+            
+             var company = await companyService.GetById(id);
+             
+
+            //then
+            Assert.Equal("IBM", company.Name);
+            Assert.Equal("Tom", company.EmployeesDto[0].Name);
+            Assert.Equal("180", company.ProfileDto.CertId);
+        }
+
+        [Fact]
+        public async Task Should_delete_company_by_Id_through_service()
+        {
+            //given
+            var context = GetCompanyDbContext();
+            CompanyDto companyDto = new CompanyDto();
+            companyDto.Name = "IBM";
+            companyDto.EmployeesDto = new List<EmployeeDto>()
+            {
+                new EmployeeDto()
+                {
+                    Name = "Tom",
+                    Age = 19,
+                }
+            };
+            companyDto.ProfileDto = new ProfileDto()
```

Comment:
can move company info  Employee and profile info to {} to make code clear and also can use List later to add more companies